dashboard/src/components/Events/Register.vue

```diff
@@ -139,54 +230,242 @@
 
 <script>
 import { LMap, LTileLayer, LMarker } from "vue2-leaflet";
+import { Icon } from 'leaflet';
+import { postJSON } from "../../helpers/Request.js";
+import { getJSON } from "../../helpers/Request.js";
+import { notification } from "@/helpers/Notifications.js";
+
+
+delete Icon.Default.prototype._getIconUrl;
+Icon.Default.mergeOptions({
+  iconRetinaUrl: require('leaflet/dist/images/marker-icon-2x.png'),
+  iconUrl: require('leaflet/dist/images/marker-icon.png'),
+  shadowUrl: require('leaflet/dist/images/marker-shadow.png'),
+});
 
 export default {
   name: "EventRegister",
+  components:{
+    LMap,
+    LTileLayer,
+    LMarker,
+  },
+  mounted() {
+    this.getCategories();
+  },
   methods: {
-    validate() {
-      if (this.$refs.loginForm.validate()) {
-        // submit form to server/API here...
+    register() {
+      if (!this.$refs.registerForm.validate()) {
+        return;
       }
+
+      this.loading = true;
+      postJSON("/events", this.model, true)
+        .then((res) => {
+          this.loading = false;
+
+          if(res.error){
+            notification({
+            message: res.message,
+            icon: "mdi-alert-circle"
+          });
+          }else{
+            notification({
+            message: res.message,
+            icon: "mdi-check-bold"
+          });
+          } 
+
+          if (!res.error) {
+            this.$router.push({ path: "/my-events" });
+          }
+        })
+        .catch((err) => {
+          this.loading = false;
+          notification({
+            message: "Ocurrió un error al hacer la petición",
+          });
+        });
     },
+    getCategories (){
+        getJSON("/categories", this.params, false)
+          .then((res) => {
+            this.categories = res.data.items;
+            console.log(res);
+          })
+          .catch((err) => {
+            console.log(err);
+          });
+      },
     reset() {
       this.$refs.form.reset();
     },
     resetValidation() {
       this.$refs.form.resetValidation();
     },
-    save(date) {
-      this.$refs.menu.save(date);
+    openDialog() {
+      if (this.showModal) {
+        this.showModal=!this.showModal
+      }
     },
-  },
-  data: () => ({
-    menu: false,
-    dialog: true,
-    tab: 0,
-    tabs: [{ name: "Registro de evento", icon: "mdi-calendar-edit" }],
-    valid: true,
-
-    nombre: "",
-    descripcion: "",
-    valor: "",
-    imagen: null,
-    fecha_nacimiento: null,
-    aforo: "",
-    select: { category: "" },
-    items: [
-      { category: "Fiesta" },
-      { category: "Deporte" },
-      { category: "Cultura" },
-      { category: "Musica" },
-      { category: "Educacion" },
-    ],
-
-    show1: false,
-    rules: {
-      required: (value) => !!value || "Campo requerido",
-      min: (v) => (v && v.length >= 8) || "Debe contener al menos 8 caracteres",
+    addMarker(event) {
+      this.markers.push(event.latlng);
```

Comment:
is it for?

Reply:
For getting coordinates by clicking the screen.
Min in rules: Deleted

Reply:
Marker pushing: Deleted